src/index.js

```diff
@@ -1,207 +1,210 @@
-import { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 
-class DidomiSDK extends Component {
-  static propTypes = {
-    apiKey: PropTypes.string,
-    iabVersion: PropTypes.number,
-    noticeId: PropTypes.string,
-    config: PropTypes.object,
-    gdprAppliesGlobally: PropTypes.bool,
-    onReady: PropTypes.func,
-    onConsentChanged: PropTypes.func,
-    onNoticeShown: PropTypes.func,
-    onNoticeHidden: PropTypes.func,
-    onNoticeBackdropclick: PropTypes.func,
-    onNoticeClickAgree: PropTypes.func,
-    onNoticeClickMoreInfo: PropTypes.func,
-    onPreferencesClickAgreeToAll: PropTypes.func,
-    onPreferencesClickDisagreeToAll: PropTypes.func,
-    onPreferencesClickPurposeAgree: PropTypes.func,
-    onPreferencesClickPurposeDisagree: PropTypes.func,
-    onPreferencesClickViewVendors: PropTypes.func,
-    onPreferencesClickSaveChoices: PropTypes.func,
-    onPreferencesClickVendorAgree: PropTypes.func,
-    onPreferencesClickVendorDisagree: PropTypes.func,
-    onPreferencesClickVendorSaveChoices: PropTypes.func
-  }
-
-  static defaultProps = {
-    apiKey: null,
-    iabVersion: 1,
-    noticeId: null,
-    config: {},
-    gdprAppliesGlobally: true,
-    onReady: () => {},
-    onConsentChanged: () => {},
-    onNoticeShown: () => {},
-    onNoticeHidden: () => {},
-    onNoticeBackdropclick: () => {},
-    onNoticeClickAgree: () => {},
-    onNoticeClickMoreInfo: () => {},
-    onPreferencesClickAgreeToAll: () => {},
-    onPreferencesClickDisagreeToAll: () => {},
-    onPreferencesClickPurposeAgree: () => {},
-    onPreferencesClickPurposeDisagree: () => {},
-    onPreferencesClickViewVendors: () => {},
-    onPreferencesClickSaveChoices: () => {},
-    onPreferencesClickVendorAgree: () => {},
-    onPreferencesClickVendorDisagree: () => {},
-    onPreferencesClickVendorSaveChoices: () => {}
-  }
-
+const DidomiSDK = ({
+  apiKey: apiKeyProp =  null,
+  iabVersion =  2,
+  noticeId =  null,
+  config =  {},
+  gdprAppliesGlobally: gdprAppliesGloballyProp =  true,
+  onReady =  () => {},
```

Comment:
if the function isn't defined by developers, the default () => {} will be used and all hooks are going to be registered, can it be really optional without defaults ?

```suggestion
  onReady,
```

Reply:
Agreed and done in https://github.com/didomi/react/pull/25/commits/1ef47ae4129cb2454d0dd86b8cc9e2453edfd681